packages/backend/src/open_payments/payment_pointer/service.ts

```diff
@@ -152,6 +161,42 @@ async function getPaymentPointer(
     .withGraphFetched('asset')
 }
 
+async function getOrRequestPaymentPointer(
+  deps: ServiceDependencies,
+  url: string
+): Promise<PaymentPointer | undefined> {
+  const existingPaymentPointer = await getPaymentPointerByUrl(deps, url)
+
+  if (existingPaymentPointer) {
+    return existingPaymentPointer
+  }
+
+  await PaymentPointerEvent.query(deps.knex).insert({
+    type: PaymentPointerEventType.PaymentPointerNotFound,
+    data: {
+      paymentPointerUrl: url
+    }
```

Comment:
We could also just return the payment pointer path here, i.e. just `gfranklin` instead of the full `https://rafiki.money/gfranklin`, but I think this keeps it consistent, we just let the ASE to the stripping of the string